app/src/main/java/dev/baseio/discordjetpackcompose/viewmodels/DashboardScreenViewModel.kt

```diff
@@ -1,27 +1,99 @@
 package dev.baseio.discordjetpackcompose.viewmodels
 
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
+import dev.baseio.discordjetpackcompose.entities.NetworkState
+import dev.baseio.discordjetpackcompose.entities.UIState
+import dev.baseio.discordjetpackcompose.entities.search.SearchSheetListItemEntity
+import dev.baseio.discordjetpackcompose.entities.server.ServerEntity
+import dev.baseio.discordjetpackcompose.usecases.GetServerUseCase
+import dev.baseio.discordjetpackcompose.usecases.search.GetSearchSheetItemListUseCase
+import dev.baseio.discordjetpackcompose.usecases.server.GetServerListUseCase
+import dev.baseio.discordjetpackcompose.utils.ioScope
+import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
-import javax.inject.Inject
 
 @HiltViewModel
-class DashboardScreenViewModel @Inject constructor() : ViewModel() {
-  private val _currentSelectedChatUsername = MutableStateFlow("")
-  val currentSelectedChatUsername: StateFlow<String> = _currentSelectedChatUsername.asStateFlow()
+class DashboardScreenViewModel @Inject constructor(
+    private val getServerUseCase: GetServerUseCase,
+    private val getSearchSheetItemListUseCase: GetSearchSheetItemListUseCase,
+    private val getServerListUseCase: GetServerListUseCase,
+): ViewModel() {
+    var currentServerEntity: UIState<ServerEntity> by mutableStateOf(UIState.Empty)
+        private set
+
+    var searchSheetItemList = mutableStateListOf<SearchSheetListItemEntity>()
+        private set
+
+    var serverList = mutableStateListOf<ServerEntity>()
+        private set
+
+    var searchSheetServerList = mutableStateListOf<ServerEntity>()
+        private set
+
+    fun getServer(serverId: String) {
+        ioScope {
+            currentServerEntity = UIState.Loading
+            currentServerEntity = when(val result = getServerUseCase(serverId = serverId)) {
+                is NetworkState.Failure -> { UIState.Failure(throwable = result.throwable) }
+                is NetworkState.Success -> { UIState.Success(data = result.data) }
+            }
+        }
+    }
+
+    fun getServers(serverIds: List<String>) {
+        serverIds.forEach { serverId ->
+            ioScope {
+                val result = getServerUseCase(serverId = serverId)
+                if (result is NetworkState.Success) {
+                    searchSheetServerList.add(result.data)
+                }
+            }
+        }
+    }
+
+    fun getSearchSheetItemList() {
+        ioScope {
```

Comment:
@shubhamsinghshubham777 we can get rid of iOscope https://developer.android.com/kotlin/coroutines/coroutines-best-practices